apps/server/src/modules/deletion/uc/deletion-execution.uc.ts

```diff
@@ -0,0 +1,22 @@
+import { Injectable } from '@nestjs/common';
+import { DeletionClient } from '../client';
+import { DeletionExecutionTriggerResult } from './interface';
+import { DeletionExecutionTriggerResultBuilder } from './builder';
+
+@Injectable()
+export class DeletionExecutionUc {
+	constructor(private readonly deletionClient: DeletionClient) {}
+
+	async triggerDeletionExecution(limit?: number): Promise<DeletionExecutionTriggerResult> {
+		// Try to trigger the deletion execution(s) via Deletion API client,
+		// return successful status in case of a success, otherwise return
+		// a result with a failure status and a proper error message.
+		try {
+			await this.deletionClient.executeDeletions(limit);
+
+			return DeletionExecutionTriggerResultBuilder.buildSuccess();
+		} catch (err) {
+			return DeletionExecutionTriggerResultBuilder.buildFailure(err as Error);
```

Comment:
I think UC should throw error, and your console api should catch it and builds result. Because if you use this UC in HTTP API, you throw error to the client

Reply:
Done in https://github.com/hpi-schul-cloud/schulcloud-server/pull/4557/commits/f269a642199f948380cdaa1c461c99b19f0ee989 .